tests/protocols/guidance_test.py

```diff
@@ -0,0 +1,178 @@
+
+	INPUT2 = [['1','3'],['0','4'],['1','0','100'],['4','-2','0']]
+	TG2MSG = [["","Reteach: Addition"],["","Reteach: Addition"],["","","Reteach: Addition"],["","","Reteach: AdditionReteach: Negation"]]
+	def setUp(self):
+		self.cmd_args = mock.Mock()
+		self.assignment = mock.Mock()
+		self.proto = unlock.protocol(self.cmd_args, self.assignment)
+		self.proto.current_test = self.TEST
+		self.proto._verify = self.mockVerify
+		self.proto._input = self.mockInput
+		self.proto.printed_msg = ""
+
+		self.input_choices = []
+		self.choice_number = 0
+
+	def mockInput(self, prompt):
+		self.choice_number += 1
+		return self.input_choices[self.choice_number - 1]
+
+	def mockVerify(self, guess, locked):
+		return guess == locked
+
+	def checkNumberOfAttempts(self, number_of_attempts):
+		self.assertEqual(number_of_attempts, len(self.proto.analytics))
+
+	def checkDictField(self, dictionary, field, expected_value):
+		self.assertIn(field, dictionary)
+		self.assertEqual(expected_value, dictionary[field])
+
+	def callsInteractError(self, expected_error, answer, choices=None, unique_id=None,
+			case_id=None, prompt=None, randomize=True):
+		if not unique_id:
+			unique_id = self.UNIQUE_ID
+		if not case_id:
+			case_id = self.CASE_ID
+		if not prompt:
+			prompt = self.PROMPT
+		if not choices:
+			choices = None
+
+		self.assertRaises(expected_error, self.proto.interact,
+						  unique_id, case_id, prompt, answer)
+
+	def callsInteract(self, expected, answer, choices=None, unique_id=None,
+				case_id=None, prompt=None, randomize=True):
+		if not unique_id:
+			unique_id = self.UNIQUE_ID
+		if not case_id:
+			case_id = self.CASE_ID
+		if not prompt:
+			prompt = self.PROMPT
+		if not choices:
+			choices = None
+
+		self.assertEqual(expected, self.proto.interact(unique_id, case_id,
+						 prompt, answer, choices=choices, randomize=randomize))
+
+	def validateRecord(self, record, answer, correct, prompt=None,
+					   unique_id=None, case_id=None, guidance_msg=None):
+		if not unique_id:
+			unique_id = self.UNIQUE_ID
+		if not case_id:
+			case_id = self.CASE_ID
+		if not prompt:
+			prompt = self.PROMPT
+		self.checkDictField(record, 'prompt', prompt)
+		self.checkDictField(record, 'answer', answer)
+		self.checkDictField(record, 'correct', correct)
+		self.checkDictField(record, 'id', unique_id)
+		self.checkDictField(record, 'case_id', case_id)
+		self.checkDictField(record,'printed msg',guidance_msg)
+
+		self.assertIn('question timestamp', record)
+		self.assertIsInstance(record['question timestamp'], int)
+
+		self.assertIn('answer timestamp', record)
+		self.assertIsInstance(record['answer timestamp'], int)
+
+	def testSingleLine_immediatelyCorrect(self):
+		self.input_choices = self.ANSWER
+		self.callsInteract(self.ANSWER, self.ANSWER)
+
+		self.checkNumberOfAttempts(1)
+		attempt = self.proto.analytics[0]
+
+		self.validateRecord(attempt, answer=self.ANSWER, correct=True,guidance_msg = self.ANSWERMSG[0])
+
+	def testTreatmentGroup0(self):
```

Comment:
If the only difference between these three tests is `self.INPUT*` and `debug = *` than it would be better to move the code to a separate function that takes the necessary parameters to run the test rather than have all of this copy paste.

In addition, if my suggestion about making `get_tg` the only place the the treatment group is set, that function can be overwritten rather than have this `debug = *` code.